bgpd/bgp_updgrp_adv.c

```diff
@@ -228,6 +228,12 @@ static int group_announce_route_walkcb(struct update_group *updgrp, void *arg)
 			   afi2str(afi), safi2str(safi), ctx->dest);
 
 	UPDGRP_FOREACH_SUBGRP (updgrp, subgrp) {
+		/* withdraw stale addpath without waiting for the coalesce timer timeout.
```

Comment:
Why is this not placed in the timer pop for the coalesce timer path?  The whole point of a coalesce timer is to allow everything to settle down,  instead we don’t Coalesce in this one case?

Reply:
Or `subgroup_announce_table()` by implicitly making withdrawals of stale routes? If the coalescing timer is very high, `subgroup_coalesce_timer()` could be delayed also and the stale BGP table happens again, right?

Reply:
I meant that if we move the withdrawal stuff to coalesce timer path, we might end-up in the situation where routes won't be withdraw if coalesce time is set to a high value (e.g. `coalesce-time 4294967295`). The question would be if we need to wait for coalesce timer at all for withdrawals?

Reply:
I don't know that we should be waiting for the coalesce timer for withdraws at all ... holding off withdraws creates situations where we send traffic down a path that doesn't exist. It seems like holding off advertisements is fine, but I don't think withdraws should be held ...

The one sticky situation is where an update is also an implicit withdraw.

Reply:
I think we're waiting on this conversation to be resolved before pushing this?

Reply:
yes, we are